hwilib/_cli.py

```diff
@@ -0,0 +1,307 @@
+
+    subparsers = parser.add_subparsers(description='Commands', dest='command')
+    # work-around to make subparser required
+    subparsers.required = True
+
+    enumerate_parser = subparsers.add_parser('enumerate', help='List all available devices')
+    enumerate_parser.set_defaults(func=enumerate_handler)
+
+    getmasterxpub_parser = subparsers.add_parser('getmasterxpub', help='Get the extended public key for BIP 44 standard derivation paths. Convenience function to get xpubs given the address type, account, and chain type.')
+    getmasterxpub_parser.add_argument("--addr-type", help="Get the master xpub used to derive addresses for this address type", type=AddressType.argparse, choices=list(AddressType), default=AddressType.WIT) # type: ignore
+    getmasterxpub_parser.add_argument("--account", help="The account number", type=int, default=0)
+    getmasterxpub_parser.set_defaults(func=getmasterxpub_handler)
+
+    signtx_parser = subparsers.add_parser('signtx', help='Sign a PSBT')
+    signtx_parser.add_argument('psbt', help='The Partially Signed Syscoin Transaction to sign')
+    signtx_parser.set_defaults(func=signtx_handler)
+
+    getxpub_parser = subparsers.add_parser('getxpub', help='Get an extended public key')
+    getxpub_parser.add_argument('path', help='The BIP 32 derivation path to derive the key at')
+    getxpub_parser.set_defaults(func=getxpub_handler)
+
+    signmsg_parser = subparsers.add_parser('signmessage', help='Sign a message')
+    signmsg_parser.add_argument('message', help='The message to sign')
+    signmsg_parser.add_argument('path', help='The BIP 32 derivation path of the key to sign the message with')
+    signmsg_parser.set_defaults(func=signmessage_handler)
+
+    getkeypool_parser = subparsers.add_parser('getkeypool', help='Get JSON array of keys that can be imported to Syscoin Core with importmulti')
+    kparg_group = getkeypool_parser.add_mutually_exclusive_group()
+    kparg_group.add_argument('--keypool', action='store_true', dest='keypool', help='Indicates that the keys are to be imported to the keypool', default=True)
+    kparg_group.add_argument('--nokeypool', action='store_false', dest='keypool', help='Indicates that the keys are not to be imported to the keypool', default=False)
+    getkeypool_parser.add_argument('--internal', action='store_true', help='Indicates that the keys are change keys')
+    kp_type_group = getkeypool_parser.add_mutually_exclusive_group()
+    kp_type_group.add_argument("--addr-type", help="The address type (and default derivation path) to produce descriptors for", type=AddressType.argparse, choices=list(AddressType), default=AddressType.WIT) # type: ignore
+    kp_type_group.add_argument('--all', action='store_true', help='Generate addresses for all standard address types (default paths: ``m/{44,49,84}h/0h/0h/[0,1]/*)``')
```

Comment:
for bip32 paths we should be type 57 coin not 0, this should be reflective everywhere in the repo